tests/baselines/reference/computedPropertyNamesTemplateLiteralTypes.types

```diff
@@ -0,0 +1,317 @@
+
+};
+
+const obj6 = {
+>obj6 : { [x: `foo${string}`]: string | boolean; [x: `foobar${string}`]: string; other: number; }
+>{  [pattern1]: true,  [pattern2]: "hello",  other: 100,} : { [x: `foo${string}`]: string | boolean; [x: `foobar${string}`]: string; other: number; }
+
+  [pattern1]: true,
+>[pattern1] : boolean
+>pattern1 : `foo${string}`
+>true : true
+
+  [pattern2]: "hello",
+>[pattern2] : string
+>pattern2 : `foobar${string}`
+>"hello" : "hello"
+
+  other: 100,
+>other : number
+>100 : 100
+
+};
+
+const obj7 = {
+>obj7 : { [x: `foo${string}`]: string | number | boolean; [x: `foobar${string}`]: string; fooooooooo: number; }
+>{  [pattern1]: true,  [pattern2]: "hello",  fooooooooo: 100,} : { [x: `foo${string}`]: string | number | boolean; [x: `foobar${string}`]: string; fooooooooo: number; }
+
+  [pattern1]: true,
+>[pattern1] : boolean
+>pattern1 : `foo${string}`
+>true : true
+
+  [pattern2]: "hello",
+>[pattern2] : string
+>pattern2 : `foobar${string}`
+>"hello" : "hello"
+
+  fooooooooo: 100,
+>fooooooooo : number
+>100 : 100
+
+};
+
+const obj8 = {
+>obj8 : { [x: `foo${string}`]: string | number | boolean; [x: `foobar${string}`]: string | number; foobarrrrr: number; }
+>{  [pattern1]: true,  [pattern2]: "hello",  foobarrrrr: 100,} : { [x: `foo${string}`]: string | number | boolean; [x: `foobar${string}`]: string | number; foobarrrrr: number; }
+
+  [pattern1]: true,
+>[pattern1] : boolean
+>pattern1 : `foo${string}`
+>true : true
+
+  [pattern2]: "hello",
+>[pattern2] : string
+>pattern2 : `foobar${string}`
+>"hello" : "hello"
+
+  foobarrrrr: 100,
+>foobarrrrr : number
+>100 : 100
+
+};
+
+const obj9 = {
+>obj9 : { [x: `foo${string}`]: string | boolean; }
+>{  [pattern1]: true,  [samepattern1]: "hello",} : { [x: `foo${string}`]: string | boolean; }
+
+  [pattern1]: true,
+>[pattern1] : boolean
+>pattern1 : `foo${string}`
+>true : true
+
+  [samepattern1]: "hello",
+>[samepattern1] : string
+>samepattern1 : `foo${string}`
+>"hello" : "hello"
+
+};
+
+const obj10 = {
+>obj10 : { readonly [x: `foo${string}`]: true; }
+>{  [pattern1]: true,} as const : { readonly [x: `foo${string}`]: true; }
+>{  [pattern1]: true,} : { readonly [x: `foo${string}`]: true; }
+
+  [pattern1]: true,
+>[pattern1] : true
+>pattern1 : `foo${string}`
+>true : true
+
+} as const;
+
+const obj11 = {
+>obj11 : { [x: `foo${string}`]: string | number | boolean; }
+>{  [pattern1]: 100,  ...obj9,} : { [x: `foo${string}`]: string | number | boolean; }
+
+  [pattern1]: 100,
+>[pattern1] : number
+>pattern1 : `foo${string}`
+>100 : 100
+
+  ...obj9,
+>obj9 : { [x: `foo${string}`]: string | boolean; }
+
+};
+
+const obj12 = {
+>obj12 : { [x: `foo${string}`]: string | number | boolean; }
+>{  ...obj9,  [pattern1]: 100,} : { [x: `foo${string}`]: string | number | boolean; }
+
+  ...obj9,
+>obj9 : { [x: `foo${string}`]: string | boolean; }
+
+  [pattern1]: 100,
+>[pattern1] : number
+>pattern1 : `foo${string}`
+>100 : 100
+
+};
+
+const obj13 = {
+>obj13 : {}

```

Comment:
It's worth noting here that this doesn't preserve any index signatures because `getUnionIndexInfos([left, right])` in `getSpreadType` doesn't preserve any index signatures that are not common to all of the input types. That feels like a separate issue to be solved - if it is meant to be solved.